classes/document.php

```diff
@@ -293,6 +293,33 @@ public function export_for_webservice() {
         return $data;
     }
 
+    public function set_data_from_engine($docdata) {
+        $fields = static::$requiredfields + static::$optionalfields + static::$enginefields;
+        foreach ($fields as $fieldname => $field) {
+
+            // Optional params might not be there.
+            if (isset($docdata[$fieldname])) {
+                if ($field['type'] === 'tdate') {
+                    // Time fields may need a preprocessing.
+                    $this->set($fieldname, static::import_time_from_engine($docdata[$fieldname]));
+                } else {
+                    // No way we can make this work if there is any multivalue field.
+                    if (is_array($docdata[$fieldname])) {
+                        throw new \core_search\engine_exception('multivaluedfield', 'search_solr', '', $fieldname);
```

Comment:
Not going to do full review, but probably not a good idea to use a lang string from search_solr

Reply:
That was from core, I must have missed it. This is a copy past from the parent document class, with a small change.

Reply:
Im not sure if that would even be necessary in Elastic, I haven't seen any multivalue fields, not sure if it something the engine outputs